operators/advanced_network/advanced_network/managers/dpdk/adv_network_dpdk_mgr.cpp

```diff
@@ -1371,156 +1369,152 @@ int DpdkMgr::rx_core_multi_q_worker(void* arg) {
                     rte_socket_id());
 
   std::array<int, Manager::MAX_RX_Q_PER_CORE> nb_rx{};
-  std::array<int, Manager::MAX_RX_Q_PER_CORE> to_copy{};
   std::array<int, Manager::MAX_RX_Q_PER_CORE> cur_pkt_in_batch{};
+  std::array<BurstParams*, Manager::MAX_RX_Q_PER_CORE> bursts{};
+  std::array<std::array<rte_mbuf*, DEFAULT_NUM_RX_BURST>, Manager::MAX_RX_Q_PER_CORE> mbuf_arr{};
+  std::array<uint64_t, Manager::MAX_RX_Q_PER_CORE> total_pkts{};
+  std::array<uint64_t, Manager::MAX_RX_Q_PER_CORE> last_cycles;
+  std::generate(last_cycles.begin(), last_cycles.end(), rte_get_tsc_cycles);
+
+  uint16_t cur_idx            = 0;
+  uint16_t cur_port;
+  uint16_t cur_q;
+  uint16_t cur_segs;
+  uint32_t cur_batch_size;
+  uint64_t cur_timeout_cycles;
+
+  auto update_cur_idx = [&]() {
+    cur_idx            = (cur_idx + 1) % num_queues;
+    cur_port           = tparams->q_params[cur_idx].port;
+    cur_q              = tparams->q_params[cur_idx].queue;
+    cur_segs           = tparams->q_params[cur_idx].num_segs;
+    cur_batch_size     = tparams->q_params[cur_idx].batch_size;
+    cur_timeout_cycles = tparams->q_params[cur_idx].timeout_us;
+  };
 
-  uint16_t cur_idx        = 0;
-  uint16_t cur_port       = tparams->q_params[cur_idx].port;
-  uint16_t cur_q          = tparams->q_params[cur_idx].queue;
-  uint16_t cur_segs       = tparams->q_params[cur_idx].num_segs;
-  uint32_t cur_batch_size = tparams->q_params[cur_idx].batch_size;
-
+  update_cur_idx();
 
   //
   //  run loop
   //
   while (!force_quit.load()) {
-    if (rte_mempool_get(tparams->meta_pool, reinterpret_cast<void**>(&bursts[cur_idx])) < 0) {
-      HOLOSCAN_LOG_ERROR("Processing function falling behind. No free buffers for metadata!");
-      exit(1);
-    }
-
-    BurstParams* burst  = bursts[cur_idx];
```

Comment:
why are you calling `bursts[cur_idx]` everytime below instead of keeping this and calling `burst` ?

Reply:
This was just a copy/paste left over. I will get rid of the burst line.